processing/src/main/java/org/apache/druid/segment/NestedDataColumnIndexer.java

```diff
@@ -411,7 +445,7 @@ public Object getObject()
         if (0 <= dimIndex && dimIndex < dims.length) {
           final StructuredData data = (StructuredData) dims[dimIndex];
           if (data != null) {
-            return data.getValue();
+            return ExprEval.bestEffortOf(data.getValue()).value();
```

Comment:
Is this counting on coercion or something?

Reply:
yeah, i did this to make it consistent with the value that is stored in the dictionary (since the selector is used for merging/persisting)